### PR TITLE
add private git repository / private image registry support docs

### LINKED ISSUE #324: update docs for private git and image registry support
Private source support is added in #285 and #81.  The docs must be updated to describe the configuration for the support.

### PR DESCRIPTION
**Pull request by akihikokuroda**
Signed-off-by: akihikokuroda <akuroda@us.ibm.com>

Add private image registry and private git repository support in docs

Closes #324

### CODE REVIEW

File: docs/git-bundles.md [hunk 1]
```diff
@@ -94,3 +91,42 @@ spec:
       repository: https://github.com/exdx/combo-bundle
   provisionerClassName: core.rukpak.io/plain
 ```
+
+## Private git repositories
+
+A Bundle can reference content in a private git repository using HTTPS protocol by creating a secret in the namespace that the provisioner is deployed.
+The secret is expected to contain `data.username` and `data.accesstoken` for the username and personal access token, respectively.
+The personal access token can be generated in the github settings.  ("Setting" -> "Developer settings" -> "Personal access tokens")
```
Comment (timflannagan) [inline docs/git-bundles.md:99]: Would it make sense to move this header above the Examples header? Thoughts?

Comment (timflannagan) [reply docs/git-bundles.md:99]: Also should we link out to https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token?

Comment (akihikokuroda) [reply docs/git-bundles.md:99]: It depends how many people want to set the credential. It probably helps the rate limit issues avoidance as well as the private repository access.

Comment (akihikokuroda) [reply docs/git-bundles.md:99]: I'll add the link.  Thanks!

Comment (akihikokuroda) [reply docs/git-bundles.md:99]: I'll move this section up.

File: docs/plain-bundle-spec.md [hunk 2]
```diff
@@ -161,3 +161,46 @@ spec:
...
+```
+
+2. Push the image to the private registry
+
+```bash
+docker tag quay.io/operator-framework/rukpak:example quay.io/my-registry/rukpak:example
+docker push quay.io/my-registry/rukpak:example
+```
+
+3. create the secret for quay.io retistry
```
Comment (tylerslaton) [inline docs/plain-bundle-spec.md:184]: nit:
```suggestion
3. Create the secret for quay.io registry
```

File: docs/plain-bundle-spec.md [hunk 3]
```diff
@@ -161,3 +161,46 @@ spec:
   provisionerClassName: core.rukpak.io/plain
 EOF
 ```
+
+## Private image registries
+
+A Bundle can reference content in a private image registry by creating an imagePullSecret in the namespace that the provisioner is deployed.
```
Comment (tylerslaton) [inline docs/plain-bundle-spec.md:167]: nit: Is the intent here to reference the exact key that is used in the spec?
```suggestion
A Bundle can reference content in a private image registry by creating an `pullSecret` in the namespace that the provisioner is deployed.
```

File: docs/git-bundles.md [hunk 4]
```diff
@@ -21,8 +21,44 @@ When creating a Bundle from a git source, a reference to a particular commit, ta
 provisioner to know where the bundle content is stored in the repository. Only one can be specified, and it is expected
 that the manifests are present in the particular commit/tag/branch at the directory specified.
 
-Currently only publicly-available git repositories are available as backing sources to Bundles. Authentication and using
-private git repositories as Bundle sources is on the roadmap.
+## Private git repositories
+
+A Bundle can reference content in a private git repository using HTTPS protocol by creating a secret in the namespace that the provisioner is deployed.
```
Comment (tylerslaton) [inline docs/git-bundles.md:26]: nit:
```suggestion
A Bundle can reference content in a private git repository using HTTPS by creating a secret in the namespace that the provisioner is deployed.
```

**Review by tylerslaton — APPROVED** [review]
Couple of nits, but nothing majorly blocking.

File: docs/plain-bundle-spec.md (hunk 2 above)
Comment (akihikokuroda) [reply docs/plain-bundle-spec.md:184]: Thanks! I fix it.

File: docs/plain-bundle-spec.md (hunk 3 above)
Comment (akihikokuroda) [reply docs/plain-bundle-spec.md:167]: Thanks! I change it.

File: docs/git-bundles.md (hunk 4 above)
Comment (akihikokuroda) [reply docs/git-bundles.md:26]: Thanks! I fix it.

File: docs/git-bundles.md [hunk 5]
```diff
@@ -21,8 +21,44 @@ When creating a Bundle from a git source, a reference to a particular commit, ta
...
+
+1. Create the secret
+
+```sh
+echo -n 'user name' > username.txt
+echo -n 'access token' > accesstoken.txt
+kubectl create secret generic gitsecret --from-file=username=./username.txt --from-file=accesstoken=./accesstoken.txt -n rukpak-system
+```
+
+2. Create a private repositry (private-registry/combo) and copy operator-framework/combo contents into it
```
Comment (exdx) [inline docs/git-bundles.md:41]:
```suggestion
2. Create a private repository (private-registry/combo) and copy operator-framework/combo contents into it
```

File: docs/plain-bundle-spec.md [hunk 6]
```diff
@@ -161,3 +161,46 @@ spec:
...
+```
+
+2. Push the image to the private registry
+
+```bash
+docker tag quay.io/operator-framework/rukpak:example quay.io/my-registry/rukpak:example
+docker push quay.io/my-registry/rukpak:example
+```
+
+3. Create the secret for quay.io retistry
```
Comment (exdx) [inline docs/plain-bundle-spec.md:184]:
```suggestion
3. Create the secret for quay.io registry
```

**Review by exdx — APPROVED** [review]
Looks good, thank you for putting together these docs

File: docs/plain-bundle-spec.md (hunk 6 above)
Comment (akihikokuroda) [reply docs/plain-bundle-spec.md]: Thanks!  I fix it.

**Pull request merged**